src/init.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2010-2020, Tarantool AUTHORS, please see AUTHORS file.
+ *
+ * Redistribution and use in source and binary forms, with or
+ * without modification, are permitted provided that the following
+ * conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above
+ *    copyright notice, this list of conditions and the
+ *    following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above
+ *    copyright notice, this list of conditions and the following
+ *    disclaimer in the documentation and/or other materials
+ *    provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY <COPYRIGHT HOLDER> ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * <COPYRIGHT HOLDER> OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+ * BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+#include <lua/utils.h>
+
+#include "merger/merger.h"
+#include "key_def/key_def.h"
+
+LUA_API int
+luaopen_mergerx(struct lua_State *L)
+{
+	/* Export C functions to Lua. */
+	static const struct luaL_Reg meta[] = {
+		{NULL, NULL}
+	};
+	luaL_register(L, "mergerx", meta);
+
+	lua_init_mergerx_merger(L);
+	lua_init_mergerx_key_def(L);
+
+	return 3;
```

Comment:
I looked over [Programming in Lua :: C Libraries](https://www.lua.org/pil/26.2.html) and bunch of another sources: I don't find any guarantees and about behaviour of `luaopen_*()` function, which returns something other than 1. It is even hard to define a correct behaviour for this case. If you want to return "mergerx" table, just do:

```
luaL_register(<...>);
lua_init_<...>(L);
lua_pop(L, 1);
lua_init_<...>(L);
lua_pop(L, 1);
return 1;
```

There is no need to lean on undocumented behaviour and do this strange `return 3;` here.

Reply:
I wanted to complain that last time I've checked with `lua_pop` it was not working and was returning string instead of object at the time of requiring `mergerx`, but now I've checked it, and it's working the same way. So returned `lua_pop(L, 1)` at the end of initialization functions.